### ISSUE TITLE
Look like incorrect result in assembling push numeric value in intel assembler

### ISSUE THREAD

**Issue by 4144** [swojtasiak/fcml-lib#330]
command from example dir: ``./fcml_asm -asa32 -osa32 -m32 "push 80h"``

Return this:
```
Number of the assembled instructions: 2
Instruction: 1
 Code: 6880000000
 Code length: 5
Instruction: 2
 Code: 666a80
 Code length: 3
Best instruction chosen by the assembler: 2
```

But ``666a80`` look like wrong.
This is bug? or exists way to give to push information about argument size?

**Comment by swojtasiak** [comment]
This behavior is 100% correct, as 080h is a positive value which cannot be encoded using imm8. So, if you'd like to get `6a80` you'd have to explicitly mark the operand as a negative one:

```
./fcml_asm -asa32 -osa32 -m32 "push -80h"
Number of the assembled instructions: 2
Instruction: 1
 Code: 6880ffffff
 Code length: 5
Instruction: 2
 Code: 6a80
 Code length: 2
Best instruction chosen by the assembler: 2
```

**Comment by 4144** [comment]
this push do 16 bit push on 32 bit system.

how to use ``push 0x80`` for 32 bit push operation? May be push32 or something like this?
i have this issue in assembler. if use normal 32 bit code, this small push corrupting stack

**Comment by swojtasiak** [comment]
So, please correct me if I'm wrong, because I'm not sure if I understand the problem. Would you like to have the first instruction to be marked as the one chosen by the assembler as the preferred one? Something like this:

```
Number of the assembled instructions: 2
Instruction: 1
 Code: 6880000000
 Code length: 5
Instruction: 2
 Code: 666a80
 Code length: 3
Best instruction chosen by the assembler: 1
```

**Comment by 4144** [comment]
All options in assembler or in command line tool set flags to use 32 bit code

if in sample assembler code
```
...
push var
...
```

as result some push commands will change esp to value 4, and some push commands if var == 0x80 will change esp to 2 bytes.

i think 32 bit code in push commands must always change esp to 4 bytes?

**Comment by swojtasiak** [comment]
Yes, I think that it makes perfect sense, the instruction generated for the default 32-bit operand size should be the one preferred here. I'll extend the instruction chooser to respect such cases.

**Comment by swojtasiak** [comment]
The fix has been merged with master, 1.3.0 will be released soon.
